test/api-binder.ts

```diff
@@ -0,0 +1,44 @@
+import { expect } from 'chai';
+import { APIBinder, APIBinderConstructOpts } from '../src/api-binder';
+import { DeviceApplicationState } from '../src/types/state';
+
+describe.only('APIBinder', () => {
+	let apiBinder: APIBinder;
+
+	before(() => {
+		apiBinder = new APIBinder({} as APIBinderConstructOpts);
+	});
+
+	describe('stripDeviceStateInLocalMode', () => {
+		const sampleState = {
+			local: {
+				ip_address: '192.168.1.42 192.168.1.99',
+				api_port: 48484,
+				api_secret:
+					'20ffbd6e15aba827dca6381912d6aeb6c3a7a7c7206d4dfadf0d2f0a9e1136',
+				os_version: 'balenaOS 2.32.0+rev4',
+				os_variant: 'dev',
+				supervisor_version: '9.16.3',
+				provisioning_progress: null,
+				provisioning_state: '',
+				status: 'Idle',
+				logs_channel: null,
+				apps: {},
+				is_on__commit: 'whatever',
+			},
+			dependent: { apps: {} },
+		} as DeviceApplicationState;
```

Comment:
@CameronDiver @Page- This is an example of state object collected on my machine.
I'm a bit worried by the fact it does not really match `DeviceApplicationState` interface. Although it all works, `DeviceApplicationState` has `config`, which, to my understanding, is data dictated by the backend. So, I'm not sure it's the best idea to use this interface for data originating from both backend and device.

Reply:
Any thoughts here?

Reply:
This is fine, the `config` field will only be present when the supervisor wants to let the api that that state has changed.